crates/telio-network-monitors/src/apple.rs

```diff
@@ -0,0 +1,118 @@
+//! Module to monitor network changes in Apple
+
+use network_framework_sys::{
+    nw_interface_get_name, nw_interface_t, nw_path_monitor_create, nw_path_monitor_set_queue,
+    nw_path_monitor_start, nw_path_monitor_t, nw_path_t,
+};
+use parking_lot::Mutex;
+use std::{
+    cell::RefCell,
+    ffi::{c_long, c_void, CStr},
+    rc::Rc,
+};
+use telio_utils::{telio_log_info, telio_log_warn};
+
+use crate::monitor::PATH_CHANGE_BROADCAST;
+
+/// Dispatch queue priority as high priority queue
+pub const DISPATCH_QUEUE_PRIORITY_HIGH: c_long = 2;
+/// Dispatch queue priority as default queue
+pub const DISPATCH_QUEUE_PRIORITY_DEFAULT: c_long = 0;
+/// Dispatch queue priority as low priority queue
+pub const DISPATCH_QUEUE_PRIORITY_LOW: c_long = -2;
+/// Dispatch queue priority as background queue
+pub const DISPATCH_QUEUE_PRIORITY_BACKGROUND: c_long = -1 << 15;
+
+/// Vector containing interface names in OS order
+pub static INTERFACE_NAMES_IN_OS_PREFERENCE_ORDER: Mutex<Vec<String>> = Mutex::new(Vec::new());
```

Comment:
Is this similar to `pub static LOCAL_ADDRS_CACHE: Lazy<Arc<StdMutex<Vec<if_addrs::Interface>>>>` from `crates/telio-network-monitors/src/monitor.rs`, if so why do we want to keep two copies of the same information in memory?